onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -2138,10 +2138,20 @@
     return false;
   }
 
+  // Here in GatherOpSupportChecker::IsOpSupportedImpl, we removed the restriction that 2nd input "indices" must be an initializer
+  // to accommodate the support for some models such as modibleBERT.
+  // However, for other cases(such as int64 type), we still require indices of gather to be an initializer
```

Comment:
nit: modibleBERT -> mobileBERT

Just checking my understanding. If the following correct?

 - it doesn't need to be an initializer for int32 as NNAPI Gather uses int32 for indices so the type matches. 
  - we require indices of other types to be an initializer as we convert the data to int32 during model building. 
  - we could potentially support non-initializer inputs for the other types if we inserted a cast.

Reply:
Thanks for summarizing it into these three clear points. 

All of the above statements are correct:)